File: 0776-split-bst/0776-split-bst.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<TreeNode*> splitBST(TreeNode* root, int target) {
        vector<TreeNode*> tree(2,NULL);
        if(!root) return {NULL,NULL};
        
        if(root->val <= target) {
            tree[0] = root;
            vector<TreeNode*> rAns = splitBST(root->right,target);
            root->right = rAns[0];
            tree[1] = rAns[1];
        }
        else if(root->val > target) {
            tree[1] = root;
            vector<TreeNode*> lAns = splitBST(root->left,target);
            root->left = lAns[1];
            tree[0] = lAns[0];
        }
        return tree;
    }
};

File: 0776-split-bst/0776-split-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode[] splitBST(TreeNode root, int target) {
        TreeNode[] res = new TreeNode[2];
        if(root==null) return res;
        if(root.val<=target) {
            res[0] = root;
            TreeNode[] lAns = splitBST(root.right,target);
            root.right = lAns[0];
            res[1] = lAns[1];
        }
        else {
            res[1] = root;
            TreeNode[] rAns = splitBST(root.left,target);
            root.left = rAns[1];
            res[0] = rAns[0];
        }
        return res;
    }
}